### ISSUE TITLE
Code to extract track title and timestamp is not perfect on tracks with dashes

### ISSUE THREAD

**Issue by crisbal** [crisbal/album-splitter#49]
As reported by @nicolaasjan in https://github.com/crisbal/album-splitter/issues/22

> However, the -'s in the filenames in the case of video https://www.youtube.com/watch?v=WUNipoTGhMc were still omitted, e.g.:
> 01 Olivia Newton-JohnMagic.wav.
> I would prefer 01 - Olivia Newton-John - Magic.wav.

This is the relevant code : https://github.com/crisbal/album-splitter/blob/master/album_splitter/parse_tracks.py#L39

Work to be done:

- Write unit tests for the current implementation
- Replace the current implementation with one easier to understand and maintain

**Comment by crisbal** [comment]
Hey @nicolaasjan

I have made some changes to the code and this is what is returned now:

```
01 Olivia Newton-John - Magic.wav
02 Rupert Holmes - Escape (The Pina Colada Song).wav
03 Captain & Tenille - Do that to me one more time.wav
04 Michael Jackson - Rock with you.wav
05 Queen - Crazy little thing called love.wav
06 Bette Midler - The rose.wav
07 Smokey Robinson - Crusin'.wav
08 Paul McCartney - Coming up.wav
09 Lipps, Inc - Funkytown.wav
10 Air Supply - Lost in love.wav
11 Elton John - Little Jeannie.wav
12 Gary Numan - Cars.wav
13 Spinners - Working my way back to you-Forgive me, girl.wav
14 KC & The Sunshine Band - Please don't go.wav
15 Styx - Babe.wav
16 Christopher Cross - Ride like the wind.wav
17 Diana Ross - Upside down.wav
18 Billy Preston & Syreeta - With you i'm born again.wav
19 Pink Floyd - Another brick in the wall (Part 2).wav
```

Please let me know if this is good for you.

There are still some edge cases, like if your `tracks.txt` contains a track number, this is treated as part of the title (For example `02 Olivia Newton-John - Magic - 00:00` will produce a track with the title `02 Olivia Newton John` also in the metadata. This is something I will try to fix later. But at least it should be much better than before.

**Comment by nicolaasjan** [comment]
> Please let me know if this is good for you.

Much better. Thanks!

Had to omit the track numbers from `tracks.txt` to get the same output you had.
Ideally, output file name should be like:
`01 - Olivia Newton-John - Magic.wav`

Also better that all is now handled in the virtual environment, so that it doesn't interfere with my own yt-dlp version. 😀️

**Comment by crisbal** [comment]
@nicolaasjan Yep, you can now `pip install` it too from the [python package index](https://pypi.org/project/album-splitter/).

**Comment by nicolaasjan** [comment]
> @nicolaasjan Yep, you can now `pip install` it too from the [python package index](https://pypi.org/project/album-splitter/).

Shouldn't there be `.wav` instead of `.mp3` in Project description ---> Downloading and splitting an album from YouTube?
